lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -54,7 +56,14 @@ def initialize(serializer, options = {})
 
         def serializable_hash(options = nil)
           options ||= {}
+          if serializer.success?
+            success_document(options)
```

Comment:
the JSON API adapter can generate either a success document or failure document. 

Right now, the switch is on the serializer method `success?` which is hard-coded in all serializers.  Alternatively, the method could be `errors?` or `failure?` or `document_type`.  

The idea is that, right now, is that the error serializer must be provided.  It is not inferred from the presence of errors on the model or any models.

eventually maybe also an 'information' document.